### DISCUSSION TITLE
Struggling with Date format in Python

### DISCUSSION THREAD

**Discussion by ArthurRedex** [atoti/atoti#901] [Q&A] [answered]
Hi
I really need some help here, I've been trying to load Panda dataframe into an Atoti Cube, but I got an error with date format whatever  I tried. 

I tried this :     
df['D_InvoiceDate'] = pd.to_datetime(df['D_InvoiceDate'],format='%Y-%m-%d')
I tried this 
df['D_InvoiceDate'] = pd.to_datetime(df['D_InvoiceDate'].values.astype('datetime64[ns]'))

But I always have the same exception: 

pydantic_core._pydantic_core.ValidationError: 1 validation error for read_pandas
1
  Input should be an instance of DataFrame [type=is_instance_of, input_value=0    2023-01-04
1    2023...e, dtype: datetime64[ns], input_type=Series]
    For further information visit https://errors.pydantic.dev/2.7/v/is_instance_of

<img width="768" alt="Screenshot 2024-07-22 at 15 09 00" src="https://github.com/user-attachments/assets/3166f927-e4dc-4eef-92b5-0ffe5cc4e2f0">

If somebody has an idea on how to simply convert Date Format (loaded as DD/MM/YYYY from an Excel file)to be Accepted by Atoti , I'm really Interested. 
Thanks ahead,
Arthur

**Comment by tibdex** [comment, marked as answer]
Hi, the error is quite clear here. `read_pandas()` expects a DataFrame (i.e. a table) but you're passing a Series (i.e. a single column).

**Reply by ArthurRedex** [reply]
Ah yes. Ok. Correct . Didn't read the error trace correctly., and focused on the wrong par tof the code. Thanks for your help
